entity-service-impl/src/main/java/org/hypertrace/entity/service/util/DocStoreConverter.java

```diff
@@ -89,8 +93,18 @@ private static Filter transform(AttributeFilter filter) {
       f.setFieldName(filter.getName());
       f.setOp(transform(filter.getOperator()));
       if (filter.hasAttributeValue()) {
-        transform(filter.getAttributeValue(), f, isPartOfAttributeMap(filter.getName()));
+        //transform(filter.getAttributeValue(), f);
+        if (ATTRIBUTES_LABELS_FIELD_NAME.equals(f.getFieldName()) && f.getOp() == Op.EQ) {
+          transformToEqFilterWithValueListRhs(filter.getAttributeValue(), f);
+        } else if (ATTRIBUTES_LABELS_FIELD_NAME.equals(f.getFieldName()) && f.getOp() == Op.IN) {
+          transformToOrFilterChainForStrArray(filter.getAttributeValue(), f);
+          // Early return on this one since the childFilters are in the Or chain.
```

Comment:
Why does this return early but none of the others? I'm guessing from the comment because this transformation involves childfilters we don't want to overwrite them later. would it be feasible the logic so it doesn't behave differently? Would there ever even be childfilters if line 95 were true or could the child filter logic move into an else clause?